app/javascript/controllers/filter_controller.js

```diff
@@ -0,0 +1,13 @@
+import { Controller } from "@hotwired/stimulus"
+import debounce from "lodash/debounce"
+
+export default class extends Controller {
+  initialize() {
+    this.submit = debounce(this.submit.bind(this), 1000)
```

Comment:
Can probably go with a lower value here
```suggestion
    this.submit = debounce(this.submit.bind(this), 300)
```